### PR TITLE
USAGOV-329-s3fs-validation: update the s3fs testbed

### PR DESCRIPTION
**Pull request by jacobaaronyeager**
## Jira Task

https://cm-jira.usa.gov/browse/USAGOV-329

## Checklist for the Developer

- [ ] A link to the JIRA ticket has been added above.
- [ ] The JIRA ticket identifies the desired result of this change.
- [ ] The JIRA ticket contains clear acceptance criteria.
- [ ] The JIRA ticket contains clear testing steps.
- [ ] The JIRA ticket contains a description of "Done".
- [ ] Any preparation/installation/update steps required for this code to work properly (drush commands, scripts, configuration updates) are documented in the ticket.

## Checklist for the Peer Reviewers
- [ ] The branch name of this PR matches the project standards.
- [ ] QA steps are followed and any changes to process are updated in the ticket.
- [ ] Code Standards are followed, there are no bad practices in use.
- [ ] Config changes (if any) include only necessary changes.
- [ ] No errors in Drupal or Client Browser.
- [ ] Code has been tested locally (if possible).
- [ ] Following AC and Testing Steps verify changes work as expected.
- [ ] There are no known side-effects outside of expected behavior.

### CODE REVIEW

File: bin/cloudgov/test/s3fs [hunk 1]
```diff
@@ -82,11 +101,37 @@ function testDrupalStreamWrapperPrivate {
 }
 
 function testDrupalStreamWrapperHTTPS {
-  echo "Expecting to read a SEMVER vX.Y.Z from $HTTPS_TEST_URL"
+  echo
+  echo "Expecting to read a SERVER vX.Y.Z from $HTTPS_TEST_URL"
```
Comment (akf) [inline bin/cloudgov/test/s3fs:106]: I think Dan really meant SEMVER: https://semver.org/ ($HTTPS_TEST_URL is a version.txt)

File: bin/cloudgov/test/s3fs-inside [hunk 2]
```diff
@@ -71,16 +85,15 @@ function testDrupalStreamWrapperPublic {
...
+  echo "Testing Drupal Stream Wrapper for private folder access.  Expecting to write and read $RUNTIME to private://$TEST_FILE_FOR_S3"
   # drupal can use the private:// stream wrapper to read and write to s3
   drush ev --root=/var/www/ "var_dump(file_put_contents('private://$TEST_FILE_FOR_S3','$RUNTIME'));"
   drush ev --root=/var/www/ "var_dump(file_get_contents('private://$TEST_FILE_FOR_S3'));"
   # independantly check for existance of the file via aws cli
 }
 
 function testDrupalStreamWrapperHTTPS {
-  echo "Expecting to read a SEMVER vX.Y.Z from $HTTPS_TEST_URL"
+  echo "Testing Drupal Stream Wrapper for HTTPS.  Expecting to read a SERVER vX.Y.Z from $HTTPS_TEST_URL"
```
Comment (akf) [inline bin/cloudgov/test/s3fs-inside:96]: SERVER -> SEMVER

**Review by akf — CHANGES_REQUESTED** [review]
I got some error messages when I ran this against dev. It looks like the first one is from testThatDrupalCanReachAWS (and probably, drupal should not be able to read AWS, since we don't have the proxy set). 

Then there's some gobbledygook because we retrieved a png and emitted it to STDOUT ... but also, it looks like it's also being interpreted as a command to run -- note the lines: 

```
/bin/bash: line 2: $'\032': command not found
/bin/bash: -c: line 3: syntax error near unexpected token `<'
```
and 

```
/bin/sh: : not found
/bin/sh: syntax error: unexpected redirection
```


```
Testing Drupal connection to s3 through Drush.

In CurlFactory.php line 200:
                                                                               
  cURL error 7: Failed to connect to aws.amazon.com port 443 after 2 ms: Coul  
  dn't connect to server (see https://curl.haxx.se/libcurl/c/libcurl-errors.h  
  tml)                                                                         
                                                                               


Testing NGINX read from s3.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  9255  100  9255    0     0  11637      0 --:--:-- --:--:-- --:--:-- 11730
/bin/bash: line 2: $'\032': command not found
/bin/bash: -c: line 3: syntax error near unexpected token `<'
IHDR?<q?tEXtSoftwareAdobe ImageReadyq?e<#iTXtXML:com.adobe.xmp<?xpacket begin="﻿" id="W5M0MpCehiHzreSzNTczkc9d"?> <x:xmpmeta xmlns:x="adobe:ns:meta/" x:xmptk="Adobe XMP Core 5.6-c140 79.160451, 2017/05/06-01:08:21        "> <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"> <rdf:Description rdf:about="" xmlns:xmp="http://ns.adobe.com/xap/1.0/" xmlns:xmpMM="http://ns.adobe.com/xap/1.0/mm/" xmlns:stRef="http://ns.adobe.com/xap/1.0/sType/ResourceRef#" xmp:CreatorTool="Adobe Photoshop CC (Macintosh)" xmpMM:InstanceID="xmp.iid:0FF1BC82B29D11E88A29AFF535B3860F" xmpMM:DocumentID="xmp.did:0FF1BC83B29D11E88A29AFF535B3860F"> <xmpMM:DerivedFrom stRef:instanceID="xmp.iid:0FF1BC80B29D11E88A29AFF535B3860F" stRef:documentID="xmp.did:0FF1BC81B29D11E88A29AFF535B3860F"/> </rdf:Description> </rdf:RDF> </x:xmpmeta> <?xpacket end="r"?>?Pb? ?IDATx??]x???e???$??? M?"?? ?????ٱ?{}XP???$H???j?$????'

Testing curl connection to s3.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  9255  100  9255    0     0  21138      0 --:--:-- --:--:-- --:--:-- 21423
/bin/sh: : not found
/bin/sh: syntax error: unexpected redirection

```

**Comment by jacobaaronyeager** [comment]
@akf All tests should be working now, though some do fail because they actually fail.

**Review by akf — APPROVED** [review]
Good enough. I get this error from "Testing NGINX read from s3" and "Testing CMS connection to S3", probably expected because we don't have the proxy set up.

`curl: (7) Failed to connect to www.usa.gov port 443 after 52 ms: Couldn't connect to server`

**Pull request merged**